pkg/controller/util/node/controller_utils.go

```diff
@@ -136,7 +136,9 @@ func MarkPodsNotReady(kubeClient clientset.Interface, pods []*v1.Pod, nodeName s
 				if !utilpod.UpdatePodCondition(&pod.Status, &cond) {
 					break
 				}
+
 				klog.V(2).Infof("Updating ready status of pod %v to false", pod.Name)
+				recorder.Eventf(pod, v1.EventTypeNormal, "NodeNotReady", "Marking for updating Pod %s unready from Node %s", pod.Name, nodeName)
```

Comment:
Can we only dispatch the event if the pod is known to still exist?  This has the event dispatched before the pod status is actually updated.  Can we dispatch an event only if we have updated the status successfully?

I think the EventType should not be Normal and instead should be Warning.

The message is redundant as the name of the pod is known (its what the event is about) and the name of the node is knowable (its where the pod is scheduled).  As a result I would update the message text to just say "Node is not ready" so we keep messages short.

Reply:
fixed